### ISSUE TITLE
Documentation for HDF5-namespace is out-of-date

### ISSUE THREAD

**Issue by anates** [dealii/dealii#8612]
The first example on https://www.dealii.org/developer/doxygen/deal.II/namespaceHDF5.html is 
```
HDF5::File data_file(filename, HDF5::File::Mode::create);
double double_attribute = 2.2;
data_file.set_attribute("double_attribute", double_attribute);
auto group = data_file.create_group("group");
group.set_attribute("simulation_type", std::string("elastic_equation"));
auto dataset = group.create_dataset<double>("dataset_name", dimensions);
dataset.set_attribute("complex_double_attribute",
                      std::complex<double>(2,2.3));
```
`HDF5::File::Mode::create` does not exist, though, it should be `HDF5::File::FileAccessMode::create`, as far as I know. I assume it is a relic from earlier versions. 
Unfortunately I do not have time for opening a PR for that myself atm, thus this issue.

**Comment by bangerth** [comment]
@dangars -- FYI.

**Comment by dangars** [comment]
Thanks @anates !
That is right, it is a relic from an earlier version.
I fixed it in #8615

**Comment by bangerth** [comment]
Thanks, everyone, for the quick turnaround!
